# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/CanvasRenderingContext2D.h
# Commit: dd00e2da3a0f
# Full Hash: dd00e2da3a0f11a7cc0af9e5ece21a339f4393eb
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 69cc1de6f754 -r dd00e2da3a0f dom/canvas/CanvasRenderingContext2D.h
--- a/dom/canvas/CanvasRenderingContext2D.h	Wed Jun 10 21:57:19 2020 +0000
+++ b/dom/canvas/CanvasRenderingContext2D.h	Wed Jun 10 22:21:02 2020 +0000
@@ -56,7 +56,6 @@
 class Optional;
 
 struct CanvasBidiProcessor;
-class CanvasRenderingContext2DUserData;
 class CanvasDrawObserver;
 class CanvasShutdownObserver;
 
@@ -87,6 +86,12 @@
     return mCanvasElement->GetOriginalCanvas();
   }
 
+  void OnBeforePaintTransaction() override;
+  void OnDidPaintTransaction() override;
+  layers::PersistentBufferProvider* GetBufferProvider() override {
+    return mBufferProvider;
+  }
+
   void Save() override;
   void Restore() override;
   void Scale(double aX, double aY, mozilla::ErrorResult& aError) override;
@@ -491,8 +496,6 @@
     }
   }
 
-  friend class CanvasRenderingContext2DUserData;
-
   virtual UniquePtr<uint8_t[]> GetImageBuffer(int32_t* aFormat) override;
 
   // Given a point, return hit region ID if it exists
@@ -720,8 +723,6 @@
 
   bool mHasPendingStableStateCallback;
 
-  nsTArray<CanvasRenderingContext2DUserData*> mUserDatas;
-
   // If mCanvasElement is not provided, then a docshell is
   nsCOMPtr<nsIDocShell> mDocShell;
 